app/controllers/concerns/course/discussion/posts_concern.rb

```diff
@@ -0,0 +1,63 @@
+module Course::Discussion::PostsConcern
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :set_topic
+    authorize_resource :discussion_topic
+    load_and_authorize_resource :post, through: :discussion_topic,
+                                       class: Course::Discussion::Post.name
+  end
+
+  def create
+    Course::Discussion::Post.transaction do
+      fail ActiveRecord::Rollback unless @post.save
+      fail ActiveRecord::Rollback unless auto_subscribe_topic
```

Comment:
You can just use `fail ActiveRecord::Rollback unless @post.save && auto_subscribe_topic`.

Maybe `auto_subscribe_topic` should be called `create_topic_subscription`